node/overseer/src/lib.rs

```diff
@@ -534,798 +319,104 @@ pub async fn forward_events<P: BlockchainEvents<Block>>(
-/// for whatever reason).
-///
-/// [`Subsystem`]: trait.Subsystem.html
-struct OverseenSubsystem<M> {
-	instance: Option<SubsystemInstance<M>>,
-}
+	#[subsystem(no_dispatch, NetworkBridgeMessage)]
+	network_bridge: NetworkBridge,
 
-impl<M> OverseenSubsystem<M> {
-	/// Send a message to the wrapped subsystem.
-	///
-	/// If the inner `instance` is `None`, nothing is happening.
-	async fn send_message(&mut self, msg: M, origin: &'static str) -> SubsystemResult<()> {
-		const MESSAGE_TIMEOUT: Duration = Duration::from_secs(10);
-
-		if let Some(ref mut instance) = self.instance {
-			match instance.tx_bounded.send(MessagePacket {
-				signals_received: instance.signals_received,
-				message: msg.into()
-			}).timeout(MESSAGE_TIMEOUT).await
-			{
-				None => {
-					tracing::error!(
-						target: LOG_TARGET,
-						%origin,
-						"Subsystem {} appears unresponsive.",
-						instance.name,
-					);
-					Err(SubsystemError::SubsystemStalled(instance.name))
-				}
-				Some(res) => res.map_err(Into::into),
-			}
-		} else {
-			Ok(())
-		}
-	}
+	#[subsystem(no_dispatch, blocking, ChainApiMessage)]
+	chain_api: ChainApi,
 
-	/// Send a signal to the wrapped subsystem.
-	///
-	/// If the inner `instance` is `None`, nothing is happening.
-	async fn send_signal(&mut self, signal: OverseerSignal) -> SubsystemResult<()> {
-		const SIGNAL_TIMEOUT: Duration = Duration::from_secs(10);
+	#[subsystem(no_dispatch, CollationGenerationMessage)]
+	collation_generation: CollationGeneration,
 
-		if let Some(ref mut instance) = self.instance {
-			match instance.tx_signal.send(signal.clone()).timeout(SIGNAL_TIMEOUT).await {
-				None => {
-					tracing::error!(
-						target: LOG_TARGET,
-						?signal,
-						received = instance.signals_received,
-						"Subsystem {} appears unresponsive.",
-						instance.name,
-					);
-					Err(SubsystemError::SubsystemStalled(instance.name))
-				}
-				Some(res) => {
-					let res = res.map_err(Into::into);
-					if res.is_ok() {
-						instance.signals_received += 1;
-					}
-					res
-				}
-			}
-		} else {
-			Ok(())
-		}
-	}
-}
+	#[subsystem(no_dispatch, CollatorProtocolMessage)]
+	collator_protocol: CollatorProtocol,
 
-#[derive(Clone)]
-struct SubsystemMeters {
-	bounded: metered::Meter,
-	unbounded: metered::Meter,
-	signals: metered::Meter,
-}
+	#[subsystem(ApprovalDistributionMessage)]
+	approval_distribution: ApprovalDistribution,
 
-impl SubsystemMeters {
-	fn read(&self) -> SubsystemMeterReadouts {
-		SubsystemMeterReadouts {
-			bounded: self.bounded.read(),
-			unbounded: self.unbounded.read(),
-			signals: self.signals.read(),
-		}
-	}
-}
+	#[subsystem(no_dispatch, ApprovalVotingMessage)]
+	approval_voting: ApprovalVoting,
 
-struct SubsystemMeterReadouts {
-	bounded: metered::Readout,
-	unbounded: metered::Readout,
-	signals: metered::Readout,
-}
+	#[subsystem(no_dispatch, GossipSupportMessage)]
+	gossip_support: GossipSupport,
 
+	#[subsystem(no_dispatch, wip, DisputeCoordinatorMessage)]
```

Comment:
`wip` :sunglasses:

Reply:
yeah, I realized that this was the case for a few subsystems, so "formalized" this into a annotation :shrug: